commands/connect/diagnose.js

```diff
@@ -46,24 +46,24 @@ module.exports = {
   needsApp: true,
   needsAuth: true,
   run: cli.command(co.wrap(function * (context, heroku) {
-    context.region = yield regions.determineRegion(context, heroku)
     let mappingResults
     let didDisplayAnything = false
     let connection = yield api.withConnection(context, heroku)
+    context.region = connection.region_url
     let results = yield cli.action('Diagnosing connection', co(function * () {
       let url = '/api/v3/connections/' + connection.id + '/validations'
       return yield api.request(context, 'GET', url)
     }))
 
     cli.log() // Blank line to separate each section
     cli.styledHeader(`Connection: ${connection.name || connection.internal_name}`)
-    if (shouldDisplay(results.json, context.flags)) {
+    if (shouldDisplay(results.data, context.flags)) {
```

Comment:
`axios` has a different schema for all response bodies.
```js
{
  // `data` is the response that was provided by the server
  data: {},
  // `status` is the HTTP status code from the server response
  status: 200,
  // `statusText` is the HTTP status message from the server response
  statusText: 'OK'
}
```

https://github.com/axios/axios#response-schema